src/components/PlaidLink/index.js

```diff
@@ -5,13 +5,15 @@ import Log from '../../libs/Log';
 
 const PlaidLink = (props) => {
     const [isPlaidLoaded, setIsPlaidLoaded] = useState(false);
-    const onSuccess = useCallback((publicToken, metadata) => {
-        props.onSuccess({publicToken, metadata});
-    }, []);
+    const onSuccess = props.onSuccess;
+    const onError = props.onError;
+    const successCallback = useCallback((publicToken, metadata) => {
+        onSuccess({publicToken, metadata});
+    }, [onSuccess]);
```

Comment:
Ah so, I was following this lint rule here:

>React Hook useCallback has a missing dependency: 'props'. Either include it or remove the dependency array. However, 'props' will change when *any* prop changes, so the preferred fix is to destructure the 'props' object outside of the useCallback call and refer to those specific props inside useCallback.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)

What do you think?

Reply:
Yeah, that makes sense. It's a bit of a GOTCHA and makes me wonder if we should just start destructuring props as a more general rule? It seems to be the approach that React is promoting.

This is tangentially related, but according to [this tweet](https://twitter.com/hswolff/status/1133759319571345408) `defaultProps` will eventually be deprecated. Why? Because React wants you to do this instead:

```js
const MyComponent = ({
   propA,
   propB = 'propBDefaultValue',
}) => {
   // Component stuff in here
}
```

Reply:
Yeahhh, I think probably our rule about de-structuring is a bit outdated - but maybe a separate concern for now, or do you think we should reopen that conversation first? At the end of the day - declaring a variable or using destructuring assignment achieve the same thing.

Reply:
Chatted about this in person, but yeah I think I'm going to just resolve this for now. Agree it's not the most critical thing and we don't want to rock the boat _toooo_ much